trackintel/preprocessing/triplegs.py

```diff
@@ -145,31 +149,40 @@ def generate_trips(spts, tpls, gap_threshold=15):
     spts_tpls_no_act = spts_tpls[~spts_tpls["activity"]]
     spts_tpls_only_act = spts_tpls[spts_tpls["activity"]]
 
+    # todo: maybe move this part down to where you need it
     # add gaps as activities, to simplify id assignment later.
     gaps = pd.DataFrame(spts_tpls.loc[gap, "user_id"])
     gaps["started_at"] = spts_tpls.loc[gap, "finished_at"] + gap_threshold / 2
     gaps[["type", "activity", "new_trip"]] = ["gap", True, True]
 
     # same for user changes
     user_change = pd.DataFrame(spts_tpls.loc[condition_new_user, "user_id"])
+    # todo: if the started_at will be used for sorting then the time difference should be as small as possible
```

Comment:
It should not make any difference as we first sort by `user_id`.

Reply:
A new data "gap" is always before the first data of this specific user and therefor behind everything of the last user. The order is never corrupted as the user id stays the same. For example
| user id | timestamp |
| ------- | --------- |
| 0       | 4000      |
| 1       | 1         |

Lets add two new user "gaps".
| user id | timestamp |
| ------- | --------- |
| 0       | 4000 - 1  |
| 0       | 4000      |
| 1       | 1 - 1     |
| 1       | 1         |

Here we used 1 one as subtrahend but the value doesn't matter.

Reply:
Agree. 👍